rust/src/dns/dns.rs

```diff
@@ -701,23 +701,45 @@ impl DNSState {
     }
 }
 
+fn probe_header_validity(header : DNSHeader, rlen: usize) -> (bool, bool, bool) {
+    let opcode = ((header.flags >> 11) & 0xf) as u8;
+    if opcode >= 7 {
+        //unassigned opcode
+        return (false, false, false);
+    }
+    if header.additional_rr as usize > rlen {
+        //not enough data for additional records
+        return (false, false, false);
+    }
```

Comment:
We can end up here if `dns_parse_request` fails with incomplete, but `dns_parse_header` passes.  In the case of TCP could we just need more data here?

Reply:
Nevermind, still results in the probe returning ALPROTO_UNKNOWN, which should lead to the parser being called again when more data is available. Is that correct @victorjulien .

Reply:
> the probe returning ALPROTO_UNKNOWN, which should lead to the parser being called again when more data is available

That seems correct to me

Reply:
> We can end up here if `dns_parse_request` fails with incomplete, but `dns_parse_header` passes. In the case of TCP could we just need more data here?

It now looks to me like there was a bug in `rlen` value